src/shared/api/genomespace/gsuploadwindow.ts

```diff
@@ -0,0 +1,89 @@
+// Copyright 2013 Broad Institute, Inc.  All rights reserved.
+
+import request from 'superagent';
+
+let jsuiRoot = "https://gsui.genomespace.org/jsui/";
+
+type Callback = (data:any) => void;
+
+export function gsUploadByGet(config:{url:string, filename?:string, successCallback?:Callback, errorCallback?:Callback}) {
+    // expect an object with url, filename (optional), successCallback(optional), errorCallback (optional)
+
+    let gsUploadUrl = jsuiRoot + "/upload/loadUrlToGenomespace.html?uploadUrl=";
+    let dest = encodeURIComponent(config.url);
+    let filenameParam = "";
+    if (config['filename'] != null) {
+        filenameParam = "&fileName=" + config.filename;
+    }
+
+    let newWin:Window & any = window.open(gsUploadUrl + dest + filenameParam, "GenomeSpace Upload", "height=340px,width=550px");
```

Comment:
Yes, you're right.  This is third party code and all I wanted to do was make it compile.